client/go/outline/config/config_shadowsocks.go

```diff
@@ -233,26 +245,37 @@ func parseShadowsocksLegacyBase64URL(url url.URL) (*ShadowsocksConfig, error) {
 		// If decoding fails, return the original url with error
 		return nil, fmt.Errorf("failed to decode host string [%v]: %w", url.String(), err)
 	}
+
+	// The decoded URI doesn't follow RFC3986, so we need our own parsing. The password is expected to be plain text.
+	userInfo, host, found := cutLast(string(decoded), "@")
+	if !found {
+		return nil, errors.New("invalid user info")
+	}
+	cipherName, secret, found := strings.Cut(userInfo, ":")
+	if !found {
+		return nil, errors.New("invalid cipher info: no ':' separator")
+	}
+
 	var fragment string
 	if url.Fragment != "" {
 		fragment = "#" + url.Fragment
 	} else {
 		fragment = ""
 	}
-	newURL, err := url.Parse(strings.ToLower(url.Scheme) + "://" + string(decoded) + fragment)
+	newURL, err := neturl.Parse(strings.ToLower(url.Scheme) + "://" + host + fragment)
 	if err != nil {
 		// if parsing fails, return the original url with error
 		return nil, fmt.Errorf("failed to parse config part: %w", err)
 	}
-	// extend this check to see if decoded string contains contains other valid fields
-	if newURL.User == nil {
-		return nil, fmt.Errorf("invalid user info: %w", err)
-	}
-	cipherInfoBytes := newURL.User.String()
-	cipherName, secret, found := strings.Cut(string(cipherInfoBytes), ":")
-	if !found {
-		return nil, errors.New("invalid cipher info: no ':' separator")
+
+	// Check if both the secret and the host have query parameters. Parsing keys containing both can
+	// be ambiguous, so we disallow it altogether to avoid surprises.
+	if strings.Contains(secret, "?") && newURL.RawQuery != "" {
```

Comment:
It's not quite clear to me that this solves the ambiguity.

Perhaps we should determine that we will be greedy and always get the last `@` to determine the host name. Then people can url-encode the `@` in the query if needed.

Reply:
I agree with this getting too complicated. Yes, the greedy approach is what I am suggesting